js/src/toast.js

```diff
@@ -166,6 +183,122 @@ class Toast {
+        const { height, marginBottom } = window.getComputedStyle(toastEl)
+
+        top += (parseInt(height, 10) + parseInt(marginBottom, 10))
+        return top
+      }, this._config.positionMargin)
+
+      if (this._config.position === PositionMap.TOP_RIGHT) {
+        this._element.classList.add(PositionMap.TOP_RIGHT)
+        this._element.style.right = `${this._config.positionMargin}px`
+      } else if (this._config.position === PositionMap.TOP_LEFT) {
+        this._element.classList.add(PositionMap.TOP_LEFT)
+        this._element.style.left = `${this._config.positionMargin}px`
+      } else {
+        const leftPx = this._getMiddleToastPosition()
+
+        this._element.classList.add(PositionMap.TOP_CENTER)
+        this._element.style.left = `${leftPx}px`
+      }
+
+      this._element.style.top = `${top}px`
+      return
+    }
+
+    const bottom = toastList.reduce((bottom, toastEl) => {
+      const { height, marginTop } = window.getComputedStyle(toastEl)
+
+      bottom += (parseInt(height, 10) + parseInt(marginTop, 10))
+      return bottom
+    }, this._config.positionMargin)
+
+    if (this._config.position === PositionMap.BOTTOM_RIGHT) {
+      this._element.classList.add(PositionMap.BOTTOM_RIGHT)
+      this._element.style.right = `${this._config.positionMargin}px`
+    } else if (this._config.position === PositionMap.BOTTOM_LEFT) {
+      this._element.classList.add(PositionMap.BOTTOM_LEFT)
+      this._element.style.left = `${this._config.positionMargin}px`
+    } else {
+      const leftPx = this._getMiddleToastPosition()
+
+      this._element.classList.add(PositionMap.BOTTOM_CENTER)
+      this._element.style.left = `${leftPx}px`
+    }
+
+    this._element.style.bottom = `${bottom}px`
+  }
+
+  _repositionExistingToasts() {
+    if (!this._element.parentNode) {
+      return
+    }
+
+    const toastList = SelectorEngine.find(`.toast.${this._config.position}`, this._element.parentNode)
+
+    toastList.forEach((toastEl, index) => {
+      const toastInstance = Toast.getInstance(toastEl)
+
+      if (toastInstance.config.position.indexOf('top-') > -1) {
+        let top = toastInstance.config.positionMargin
+
+        if (index > 0) {
+          const previousToast = toastList[index - 1]
+          const { height, marginBottom } = window.getComputedStyle(previousToast)
+
+          top += (parseInt(height, 10) + parseInt(marginBottom, 10))
+        }
+
+        toastEl.style.top = `${top}px`
+      }
+
+      if (toastInstance.config.position.indexOf('bottom-') > -1) {
+        let bottom = toastInstance.config.positionMargin
+
+        if (index > 0) {
+          const previousToast = toastList[index - 1]
+          const { height, marginTop } = window.getComputedStyle(previousToast)
+
+          bottom += (parseInt(height, 10) + parseInt(marginTop, 10))
+        }
+
+        toastEl.style.bottom = `${bottom}px`
+      }
+    })
+  }
+
+  _clearPositioning() {
+    this._element.style.position = ''
```

Comment:
Isn't there any way to set multiple properties at once? Maybe with `style.cssText`? IIRC this can be quite costly because the element is repainted every time a property changes. Might be worth checking other places too later :)

Reply:
I will check that thanks 👍